app/helpers/application_helper.rb

```diff
@@ -152,13 +156,42 @@ def process_physical_holding(holding, bib_id, holding_id, is_journal, pub_date)
     info << content_tag(:ul, "#{holding_label('Location note:')} #{listify_array(holding['location_note'])}".html_safe, class: 'location-note') unless holding['location_note'].nil?
     info << content_tag(:ul, "#{holding_label('Location has:')} #{listify_array(holding['location_has'])}".html_safe, class: 'location-has') unless holding['location_has'].nil?
     info << content_tag(:ul, ''.html_safe, class: 'journal-current-issues', data: { journal: true, holding_id: holding_id }) if is_journal
+    # Label SCSB Use Restrictions
+    if /^scsb.+/ =~ holding['location_code']
+      info << recap_item_list(holding) unless holding['items'].nil?
+    end
     unless holding_id == 'thesis' && pub_date > 2012
-      info << request_placeholder(bib_id, holding_id, location_rules).html_safe
+      info << request_placeholder(bib_id, holding_id, location_rules, holding).html_safe
     end
     info = content_tag(:div, info.html_safe, class: 'holding-block') unless info.empty?
     info
   end
 
+  def recap_item_list(holding)
+    restricted_items = holding['items'].map do |item|
+      unless item['use_statement'].blank?
+        content_tag(:li, item['use_statement'].to_s)
+      end
+    end
+    restricted_items.compact!
+    if restricted_items.empty?
+      ''
+    else
+      restricted_items.uniq!
+      content_tag(:ul, "#{holding_label('Use Restrictions:')} #{restricted_items.join}".html_safe, class: 'item-list')
+    end
+  end
+
+  def recap_supervised_items?(holding)
+    if holding.key? 'items'
+      restricted_items = holding['items'].map do |item|
+        return true if item['use_statement'] == 'Supervised Use'
+      end
+      restricted_items.compact!
+      return false if restricted_items.empty?
+    end
+  end
```

Comment:
This method is sloppy. If there aren't any items in the holding it returns nil. I think it would be cleaner if it looked like this:
```ruby
  def recap_supervised_items?(holding)
    if holding.key? 'items'
      restricted_items = holding['items'].select { |item| item['use_statement'] == 'Supervised Use' }
      !restricted_items.empty?
    else
      false
    end
  end
```
The select method will populate an array for any items that match the boolean condition. If the array is empty there are no supervised items and the method returns false. Ideally this method should be tested more directly

Reply:
Longer term, checking if all these different strings match (e.g. `item['use_statement'] == 'Supervised Use'`) is going to be difficult to maintain. Seems like a file of constants might be a good idea.

Reply:
I agree generally agree that hardcoding strings like this isn't a good practice. But at the present time this is literally the only constraint.